pages/login.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useEffect, useState } from "react";
+import Head from "next/head";
+import Image from "next/image";
+import Link from "next/link";
+import styles from "@/styles/Home.module.scss";
+
+interface Input {
+  email: string;
+  password: string;
+}
+
+interface Message {
+  emailMsg: string;
+  passwordMsg: string;
+}
+
+interface Valid {
+  emailValid: boolean;
+  passwordValid: boolean;
+}
+
+const Login = () => {
+  const [input, setInput] = useState<Input>({
+    email: "",
+    password: "",
+  });
+  const [message, setMessage] = useState<Message>({
+    emailMsg: "",
+    passwordMsg: "",
+  });
+
+  const emailRegExp =
+    /^[A-Za-z0-9]([-_.]?[A-Za-z0-9])*@[A-Za-z0-9]([-_.]?[A-Za-z0-9])*\.[A-Za-z]{2,3}$/;
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInput({ ...input, [e.target.name]: e.target.value });
+  };
```

Comment:
```suggestion
  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setInput((prev)=>{ ...prev, [e.target.name]: e.target.value });
  };
```
위 코드는 제가 공유한것처럼 관리하는게 좋아요. input이 이 함수가 실행되는 시점에 100퍼센트 값이 동기화 됐다고 보장할 수 없어요. 그런데 위처럼 값을 사용하면 prev라는 값은 클로저에 의해 값이 보존되서 값을 안전하게 사용할 수 있어요.